lib/src/utils.dart

```diff
@@ -247,3 +247,13 @@ Stream<ProcessSignal> getSignals() => Platform.isWindows
     ? ProcessSignal.sigint.watch()
     : StreamGroup
         .merge([ProcessSignal.sigterm.watch(), ProcessSignal.sigint.watch()]);
+
+/// Build a list like "A, B and 3 more".
+String buildSample<T>(List<T> items, String format(T item)) {
```

Comment:
Use an Iterable<T> instead so it's not required to build a full list and only the needed elements can be built lazily.

The caller could be required to format the strings themselves, which lets you remove the second parameter and the genericness. For instance, buildSample(objectIterator.map((x) => x.toString()). This would work lazily and produce just the needed elements.

Maybe have an optional parameter with the amount of items to include? Perhaps a number of characters parameter, where it tries to fills as many items in the output while not producing too long a string.

Is sample the standard word for this feature?

Reply:
Skipped the optional parameter for amount/length for now, but did the rest.

Re: naming: I'm not sure either, but sample seems to be close enough.